### PR TITLE
fix: persist lifecycle toggles

### LINKED ISSUE #14542: Lifecycle Insight not persisting Lifecycle Toggles
## Bug description

https://user-images.githubusercontent.com/99246719/222758904-0cddd266-d879-4f06-a9a1-dd24fad4c1c1.mov

- Link to our Lifecycle Insight: https://eu.posthog.com/insights/WgqCSamz

## How to reproduce

1. "Edit" the Lifecycle Insight
2. Toggle off any of the "Lifecycle Toggles"
3. "Save" the insight - view persists with the filter applied.
4. Refresh the page - the filters have reset

## Environment

- [x] PostHog Cloud EU, project ID: 2209

## Additional context

Have also mentioned this in Slack: https://posthogusers.slack.com/archives/C01GLBKHKQT/p1677855555871769?thread_ts=1677849224.389829&cid=C01GLBKHKQT

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

From reading the code... I don't see where we persist the lifecycle toggles in the front end and I don't believe we have tests that show how they should filter on the backend. 🤷 

Closes #14542 

## Changes

* persist the lifecycle toggles onto the filters
* use insight state instead of storing it separately on trends logic
* doesn't try and filter the lifecycles by toggled state in the backend
* de-ANT the lifecycle toggles and allow the layout to change

lifecycles

## How did you test this code?

👀

### CODE REVIEW

File: frontend/src/scenes/trends/trendsLogic.ts [hunk 1]
```diff
@@ -53,18 +53,6 @@ export const trendsLogic = kea<trendsLogicType>([
...
-        toggledLifecycles: [
-            ['new', 'resurrecting', 'returning', 'dormant'],
-            {
-                toggleLifecycle: (state, { lifecycleName }) => {
-                    if (state.includes(lifecycleName)) {
-                        return state.filter((lifecycles) => lifecycles !== lifecycleName)
-                    }
-                    return [...state, lifecycleName]
-                },
-                setLifecycles: (_, { lifecycles }) => lifecycles,
```
Comment (mariusandra) [inline frontend/src/scenes/trends/trendsLogic.ts:65]: Should this `setLifecycles` still set `toggledLifecycles` like it used to? E.g. by calling setFilters or so? I think it's still being called.

File: frontend/src/scenes/insights/EditorFilters/LifecycleToggles.scss [hunk 2]
```diff
@@ -1,38 +0,0 @@
-.LifecycleToggles {
```
Comment (mariusandra) [inline frontend/src/scenes/insights/EditorFilters/LifecycleToggles.scss:1]: This file is still linked to from somewhere, causing CI to fail

**Review by mariusandra — COMMENTED** [review]
Thanks for jumping on this so fast (there was no rush though :P). Found some stuff to comment on, but didn't run the code yet. Also CC @thmsobrmlr as this might be relevant

File: frontend/src/scenes/trends/trendsLogic.ts (hunk 1 above)
Comment (pauldambra) [reply frontend/src/scenes/trends/trendsLogic.ts:65]: data-lifecycle

**Review by mariusandra — APPROVED** [review]
Well then, on to the next defining feature. Thanks for fixing this!

I'll give you the honour of also closing these with a "smashed it":
- https://github.com/PostHog/posthog/issues/12728
- https://github.com/PostHog/posthog/issues/4787

**Pull request merged**